docs/src/main/asciidoc/spring-cloud-commons.adoc

```diff
@@ -1059,3 +1059,99 @@ myrandom=${cachedrandom.appname.value}
 == Configuration Properties
 
 To see the list of all Spring Cloud Commons related configuration properties please check link:appendix.html[the Appendix page].
+
+== Spring Cloud Request Hedging
+=== Introduction
+Request hedging is a mechanism to improve performance of idempotent requests with large worst-case response times. It
+does this by automatically retrying after the original request has been active for some configurable amount of time,
+usually based on a quantile of the request's typical response time. Unlike timing out the request and retrying, request
+hedging will keep the original requests and return the first request to succeed. This trades off a small percentage
+of additional network traffic to reduce the long-tail latency.
+
+=== Usage
+You can configure a `WebClient` to do request hedging by creating a `HedgingPolicy` `@Bean` and use the `@Hedged`
+qualifier. The `HedgingPolicy` allows you to specify how long to wait before executing the extra requests, a maximum
+number of extra requests to create. Most users will want the `LatencyPercentileHedgingPolicy`, which executes new
+requests when the previous ones have exceeded some percentile of the historical latency.
+
+This example shows a typical usage:
+
+====
+[source,java,indent=0]
+----
+@Configuration
+public class MyConfiguration {
+    @Bean
+    HedgingPolicy hedgingPolicy() {
+    	return new LatencyPercentileHedgingPolicy(
+            3, // Maximum extra requests to execute
+            0.95 // Percentile of latency to wait before executing an additional request
+    	);
+    }
+
+    @Hedged(hedgingPolicy = "hedgingPolicy")
+    @Bean
+    WebClient.Builder webClientBuilder() {
+        return WebClient.builder();
+    }
+}
+
+public class MyClass {
+    @Autowired
+    private WebClient.Builder webClientBuilder;
+
+    public String doOtherStuff() {
+        return webClientBuilder.build().get().uri("http://www.example.com/")
+        				.retrieve().bodyToMono(String.class);
+    }
+}
+----
+====
+
+=== Reporting
+Hedging also provides a `HedgingMetricsReporter`. The extra requests generated by hedging could produce false positives
+in monitoring. Instances of `HedgingMetricsReporter` wired into the hedging `ExchangeFilter` will receive only the
+single request attempt that was used by the caller, ignoring any other attempts that were made and discarded.  These
+reporters are wired in to the `@Hedged` qualifier, as in this example:
+
+====
+[source,java,indent=0]
+----
+import org.springframework.cloud.client.hedger.HedgingMetricsReporter;@Configuration
```

Comment:
Nit: same line formattting?

Reply:
The stupid editor keeps trying to fix imports in this code and it doesn't even put the line break there. :frowning_face: